### PR TITLE
Convert ReactPropForShadowNodeSpecTest to Kotlin

### PR DESCRIPTION
**Pull request by IzuEneh**
## Summary:

This PR converts ReactPropForShadowNodeSpecTest to Kotlin. #38825 

## Changelog:

[INTERNAL] [CHANGED] - Kotlinify ReactPropForShadowNodeSpecTest.java

For more details, see:
https://reactnative.dev/contributing/changelogs-in-pull-requests
-->

## Test Plan:

`./gradlew :packages:react-native:ReactAndroid:test` must pass

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+ * Test that verifies that spec of methods annotated with @ReactProp in {@link ReactShadowNode} is
+ * correct
+ */
+@RunWith(RobolectricTestRunner::class)
+@Ignore
+class ReactPropForShadowNodeSpecTest {
+
+  companion object {
+    private class BaseViewManager(
+      private val mShadowNodeClass: Class<out ReactShadowNode<ReactShadowNodeImpl>>
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt:27]: Using Hungarian notation is not idiomatic for Kotlin, could you please name it `shadowNodeClass` instead?

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * Test that verifies that spec of methods annotated with @ReactProp in {@link ReactShadowNode} is
+ * correct
+ */
+@RunWith(RobolectricTestRunner::class)
+@Ignore
+class ReactPropForShadowNodeSpecTest {
+
+  companion object {
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt:25]: Could you please move the companion object to the end of the class?

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    private class BaseViewManager(
+      private val mShadowNodeClass: Class<out ReactShadowNode<ReactShadowNodeImpl>>
+    ): ViewManager<View?,ReactShadowNode<*>>() {
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt:28]: `View` could probably be non-nullable here.

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    ).nativeProps
+  }
+
+  @Test(expected = RuntimeException::class)
+  fun testGroupNoIndexParam() {
+    BaseViewManager(
+      object : ReactShadowNodeImpl() {
+        @ReactPropGroup(names = ["prop1", "prop2"])
+        fun setterWithNoIndexParam(value: Float, boolean: Boolean) {}
+      }::class.java
```
Comment (rshest) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt:99]: Why `::class.java` here vs `.javaClass` in prior clauses?

**Review by rshest — COMMENTED** [review]
Thanks a lot for working on this!

Just a couple of nitpick comments :)

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt (hunk 1 above)
Comment (IzuEneh) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt]: Updated!

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt (hunk 2 above)
Comment (IzuEneh) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt]: Moved

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt (hunk 3 above)
Comment (IzuEneh) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt]: Updated

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt (hunk 4 above)
Comment (IzuEneh) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/ReactPropForShadowNodeSpecTest.kt]: Silly mistake on my end. This has been updated

**Comment by IzuEneh** [comment]
> Thanks a lot for working on this!
> 
> Just a couple of nitpick comments :)

Hey thanks for the quick review @rshest ! I actually created this PR because I ran into a roadblock. The issue (#38825) asks us to 
1. Fix the tests and then
2. kotlinify them 

I was able to kotlinify the test but i am stuck on making it pass hence why it is still ignored.

I run into an Unsatisfied link error caused by the yoga dependency
image
I didn't want to make any changes to the gradle files so I was wandering if anyone knew the next steps to remedy this.

**Comment by rshest** [comment]
@IzuEneh I'll check it out, but otherwise we could take fixing as a separate follow-up.

**Pull request closed without merging**